src/main/java/de/tum/in/www1/artemis/service/connectors/BambooService.java

```diff
@@ -623,10 +628,11 @@ private void createAutomaticFeedback(Result result, String methodName, boolean p
      *
      * @param result the result for which the feedback should be added
      * @param jobs   the jobs list of the requestBody
-     * @return a list of feedbacks itemsstored in a result
+     * @param isStaticCodeAnalysisEnabled flag determining whether static code analysis was enabled
+     * @return a list of feedback items stored in a result
      */
     @SuppressWarnings("unchecked")
-    private Result addFeedbackToResultNew(Result result, List<BambooBuildResultNotificationDTO.BambooJobDTO> jobs) {
+    private Result addFeedbackToResultNew(Result result, List<BambooBuildResultNotificationDTO.BambooJobDTO> jobs, Boolean isStaticCodeAnalysisEnabled) {
```

Comment:
isStaticCodeAnalysisEnabled could be null which might be problematic

Reply:
I've implemented extra checks when creating, importing and editing programming exercises and secured those checks with integration tests. Additionally the database field isStaticCodeAnalysisEnabled is by default false and not nullable. This should be fine